src/app/shared/store/registration.state.ts

```diff
@@ -111,20 +120,24 @@ export class RegistrationState {
   @Action(GetProfile)
   getProfile({ patchState, getState }: StateContext<RegistrationStateModel>, { }: GetProfile) {
     const state = getState();
-
+    patchState({ role: state.user.role});
     if (state.user.role === Role.parent) {
       return this.parentService
         .getProfile()
         .pipe(
           tap(
-            (parent: Parent) => patchState({ parent: parent })
+            (parent: Parent) => {
+              patchState({ parent: parent });
+          }
           ));
     } else {
       return this.providerService
         .getProfile()
         .pipe(
           tap(
-            (provider: Provider) => patchState({ provider: provider })
+            (provider: Provider) => {
+              patchState({ provider: provider });
+          }
```

Comment:
You duplicate teh same action twice. You patch the state when you got user and then is doing it again in if/else.
I suggest to move  patchState({ role: state.user.role}); after if/else and use it only once. Or remove 123 line.